src/js/actions/ChaosActions.js

```diff
@@ -0,0 +1,103 @@
+var EventEmitter = require("events").EventEmitter;
+var lazy = require("lazy.js");
+var request = require("../helpers/qajaxWrapper");
+
+var config = require("../config/config");
+
+var AppsStore = require("../stores/AppsStore");
+var ChaosEvents = require("../events/ChaosEvents");
+
+function requestApp(appId, onSuccess, onError) {
+  request({
+    url: config.apiURL + "v2/apps/" + appId
+  })
+  .success(function (data) {
+    onSuccess(data.body.app);
+  })
+  .error(function () {
+    onError();
+  });
+}
+
+function deleteTasks(taskIds = [], onSuccess) {
+  request({
+    method: "POST",
+    data: {
+      "ids": taskIds
+    },
+    headers: {
+      "Content-Type": "application/json"
+    },
+    url: config.apiURL + "v2/tasks/delete"
+  })
+  .success(function () {
+    onSuccess();
+  })
+  .error(function (error) {
+    onSuccess();
+  });
+}
+
+var ChaosActions = lazy(EventEmitter.prototype).extend({
+  makeChaos: function (amountPercentage = 10) {
+    ChaosActions.emit(ChaosEvents.STARTED);
+
+    var apps = lazy(AppsStore.apps)
+      .filter(function (app) {
+        return app.tasksRunning > 1;
+      })
+      .shuffle();
+
+    var iterator = apps.getIterator();
+
+    var totalTasksToKill = lazy([]);
+
+    var iterateOverAppsAndMakeChaos = function () {
```

Comment:
In this large function is very hard to understand what is going on. Could we move out some of the sub-functions and use defaults and get rid of some of these nested if-else statements?